### ISSUE TITLE
Timeout if profile picture URL not reachable

### ISSUE THREAD

**Issue by robertbln** [bigbluebutton/greenlight#1760]
Since we have updated from the version 2.5.6 to 2.6.2, the Greenlight frontend runs into a timeout if you login and have set before a profile picture which URL is not reachable. Our firewall settings on the Greenlight server don't allow outgoing connections to destinations outside of the company. 

The workaround is to open all outgoing connections in the firewall or to remove all URLs in the database in the field "image" in the table "users".

**Comment by bayden10** [comment]
are you seeing the same issue with gmail profile images?
 
``` https://lh3.googleusercontent.com/a-/ ```

**Comment by robertbln** [comment]
Yes, as the firewall is configured not to accept outgoing connections, that runs also into a timeout. The user interface is not accessable anymore, becase the timeout error message comes everytime.

**Comment by farhatahmad** [comment]
We've added a check for security reasons that does a couple of checks on the photo before displaying it. It's supposed to just be blank if theres any issue loading the picture but that doesn't seem like the case based on what you're telling me

**Comment by ArminXG** [comment]
I have the same issue here. If the user by accident save a wrong URL, that user cannot use BBB any longer.

**Comment by pshunter** [comment]
Hi, I have a similar issue but I would like to confirm that first.
I have a working BBB deployment with Greenlight, but it was behind a NAT and I recently acquired a public IP so I could re-deploy a new instance without NAT.
The greenlight database was copied from the old instance. One of users (everyone is authenticated through ldap) is now unable to login on the new instance according to BBB nginx logs, there is a timeout of the Greenlight service :
```2020/11/26 15:12:36 [error] 13077#13077: *9190 upstream timed out (110: Connection timed out) while reading response header from upstream, client: [REDACTED], server: [REDACTED], request: "POST /b/auth/ldap HTTP/1.1", upstream: "http://127.0.0.1:5000/b/auth/ldap", host: "[REDACTED]", referrer: "http://[REDACTED]/b/ldap_signin"```

We tried that with different browsers and from different computers. My own account has no problems logging on from the same machine. I post here because the symptoms are similar but I don't think our users have external profile images. There are images but they are embedded (the ldap is a MS AD)

There is no error to be seen in greenlight/log/production.log

Is there a way to increase Greenlight's log output or any other place to look for debug information ?

Thanks

**Comment by pshunter** [comment]
Here is the associated snippet from production.log :
```2020-11-26 14:10:58 +0000 - INFO: [c9426335-49a1-48af-82d5-7465697439c9] method=GET path=/b/ format=html controller=MainController action=index status=200 duration=43.51 view=23.83 db=6.68 host=greenlight 
2020-11-26 14:11:00 +0000 - INFO: [e9669346-0342-46a6-b084-40eb1ded3a4b] method=GET path=/b/signin format=html controller=SessionsController action=signin status=302 duration=7.57 view=0.00 db=1.06 location=http
2020-11-26 14:11:00 +0000 - INFO: [db22b98d-ab9f-4b0a-bbc3-bbc9fb959234] method=GET path=/b/ldap_signin format=html controller=SessionsController action=ldap_signin status=200 duration=23.36 view=16.97 db=0.90 h
2020-11-26 14:11:49 +0000 - INFO: [a527ddf6-99ec-42f3-9017-129933c07417] method=GET path=/b/ldap_signin format=html controller=SessionsController action=ldap_signin status=200 duration=36.93 view=16.70 db=7.66 h```
It also somewhat resembles #2311

**Comment by robertbln** [comment]
If you have a Greenlight also on a test system, I would just try to remove all user images and check if the timeout still happens.

```
docker exec -it --user postgres greenlight-db sh
psql
\c greenlight_production
select * from users where image <> '';
update users set image = '' where image not like '';
```

**Comment by pshunter** [comment]
Ok the images are taken from the LDAP AD so removing them from the database won't help. We deleted the picture directly in the AD from the problematic account and the timeout disappeared.
Most of my users have pictures so I wonder how much will have trouble connecting.
Can somebody add some insight on what could make greenlight choke when retrieving pictures ? What about issue #841 since the issue seems to be related ? Is it really fixed ?

**Comment by pshunter** [comment]
Hi, FTR my LDAP problem had nothing to do with greenlight. A firewall was blocking large ldap packets due to big user photos because it was suspecting some form of overflow attack.
Thanks for helping to track this down.

**Comment by farhatahmad** [comment]
I believe this was fixed in v2.8

**Comment by robertbln** [comment]
thanks, I've tested it now. Works as expected.
